File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n < 3:
            return 0 if n == 0 else 1

        trib = [0, 1, 1]

        while len(trib) <= n:
            trib.append(trib[-3] + trib[-2] + trib[-1])

        return trib[-1]

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        vector<int>trib {0, 1, 1};
        for (int i = 3; i <= n; ++i) {
            int len = trib.size();
            trib.push_back(trib[len-1] + trib[len-2] + trib[len-3]);
        }
        return trib[n];
    }
};